crates/oxc_minifier/src/ast_passes/peephole_remove_dead_code.rs

```diff
@@ -379,17 +443,16 @@ mod test {
     fn test_array_literal() {
         fold("([])", "");
         fold("([1])", "");
-        // fold("([a])", "a");
-        // fold("([foo()])", "foo()");
+        fold("([a])", "a");
+        fold("([foo()])", "foo()");
     }
 
     #[test]
-    #[ignore]
     fn test_array_literal_containing_spread() {
         fold_same("([...c])");
         fold("([4, ...c, a])", "([...c])");
-        fold("([foo(), ...c, bar()])", "(foo(), [...c], bar())");
-        fold("([...a, b, ...c])", "([...a], [...c])");
+        //fold("([foo(), ...c, bar()])", "(foo(), [...c], bar())");
+        //fold("([...a, b, ...c])", "([...a], [...c])");
```

Comment:
my core currently outputs `foo(), bar(), [...c];` for the first commented test
and `[...a, ...c];` for the second

i think this is ok to just change? thoughts?

Reply:
There may be different semantics? You need to play safe 😅 

What does other minifiers compress to?

Reply:
ah yeah you're right.
spreading `c` could mutate bar

i think the our fix for the 2nd case is fine.


| input               | esbuild        | swc            | terser   | closure             |
| ------------------- | -------------- | -------------- | -------- | ------------------- |
| `([...a, b, ...c])` | `[...a,...c];` | `[...a,...c];` | `a,b,c;` | `([...a], [...c])"` |

esbuild/swc have the same behaviour. i think terser's behaviour is just wrong.